allow thin 1.6.x

mathieujobin opened the pull request:
upgrading skinny from 0.2.2 to 0.2.3 cause thin to downgrade from 1.6.3

I think ~> are a very good practice for applications Gemfiles but not so for library gemspecs
specifying minimum ( and maximum if necessary is the way to go )

sj26 commented:
Nope, given how heavily skinny digs into Thin to manipulate the eventmachine internals this gem has been kept explicitly versioned. A minor change could make skinny no longer function.

If it's working with thin 1.6 we can bump the dependency to `">= 1.5", "< 1.7"` for now.

Additionally, mailcatcher is an application in its own right. It should not be used in a Gemfile or similar where its dependencies might conflict with your applications.

mathieujobin commented:
I understand,  ">= 1.5", "< 1.7" sounds good to me.

Skinny 0.2.2 is working fine with thin 1.6.3

mathieujobin commented:
patch updated

melnikaite commented:
+1

mentero commented:
Any news here?

mathieujobin committed:
allow thin 1.6.x

upgrading skinny from 0.2.2 to 0.2.3 cause thin to downgrade from 1.6.3

I think ~> are a very good practice for applications Gemfiles but not so for library gemspecs
specifying minimum ( and maximum if necessary is the way to go )

mathieujobin committed:
update requirement on thin to allow both 1.5 and 1.6, but nothing else.

sj26 commented:
Eventmachine doesn't need to be unrestricted. And we only now guarantee 1.5 and 1.6, maybe 1.7 will do something crazy. We're not just using public API which is why the restriction is so pessimistic. This gem is a bit of a hack.

The story would be different if thin supported rack.hijack or something.

sj26 commented:
I've relaxed the thin restriction a little after some testing in 6ab07de1be58bb63dbb9626d519c782ab6725ab8 and released v0.2.4.

luckman212 commented:
Any chance there'll be an update that allows `thin` >= **1.7**?

sj26 commented:
@luckman212 there could be! It probably requires adding some testing and then bumping the version, to make sure nothing breaks.
